src/rust/engine/options/src/build_root.rs

```diff
@@ -14,8 +14,13 @@ impl BuildRoot {
   const SENTINEL_FILES: &'static [&'static str] = &["pants.toml", "BUILDROOT", "BUILD_ROOT"];
 
   pub fn find() -> Result<BuildRoot, String> {
-    let cwd = env::current_dir().map_err(|e| format!("Failed to determine $CWD: {e}"))?;
-    Self::find_from(&cwd)
+    match env::var("PANTS_BUILDROOT_OVERRIDE") {
+      Ok(buildroot) => Ok(BuildRoot(PathBuf::from(buildroot))),
```

Comment:
If `PANTS_BUILDROOT_OVERRIDE` is explicitly set to be empty, e.g. `PANTS_BUILDROOT_OVERRIDE= the-command-here`, this sets `BuildRoot(PathBuf::from(""))` which seems to be invalid. Maybe this could have "empty" behave the same as "not set"?

```suggestion
      Ok(buildroot) if !buildroot.is_empty() => Ok(BuildRoot(PathBuf::from(buildroot))),
```